src/compiler/propagation/type_propagator.cc

```diff
@@ -84,6 +84,82 @@ void TypePropagator::ensure_entry_main() {
   has_entry_main_ = true;
 }
 
+void TypePropagator::ensure_entry_spawn() {
+  if (has_entry_spawn_) return;
+  TypeScope scope(1, words_per_type());
+  TypeStack* stack = scope.top();
+  stack->push_instance(program()->task_class_id()->value());
+  Method target = program()->entry_spawn();
+  call_static(null, &scope, null, target);
+  has_entry_spawn_ = true;
+}
+
+void TypePropagator::ensure_entry_task() {
+  if (has_entry_task_) return;
+  TypeScope scope(1, words_per_type());
+  TypeStack* stack = scope.top();
+  stack->push_any();  // TODO(kasper): Should be lambda.
+  Method target = program()->entry_task();
+  call_static(null, &scope, null, target);
+  has_entry_task_ = true;
+}
+
+void TypePropagator::ensure_lookup_failure() {
+  if (has_lookup_failure_) return;
+  TypeScope scope(2, words_per_type());
+  TypeStack* stack = scope.top();
+  stack->push_any();           // receiver
+  stack->push_smi(program());  // selector_offset
```

Comment:
Maybe add a comment explaining that implicit lookup failures always have the integer selector offset.
Explicit ones may have a string but are evaluated normally anyway.

Reply:
Fixed.